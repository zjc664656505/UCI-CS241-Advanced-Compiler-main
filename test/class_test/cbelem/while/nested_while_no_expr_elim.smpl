main
var a, b;
{
    let a <- 1;
    let b <- (a + a);
    while a < 10 do
        let a <- a + 1;
        while a < b do
            let b <- b + 1;
        od;
    od;
}.
