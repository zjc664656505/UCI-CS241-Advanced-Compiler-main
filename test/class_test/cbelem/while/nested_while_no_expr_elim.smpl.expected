BB0 (constants block):

  Instructions
    0: CONST 1
    2: CONST 10

BB1:
  Program Variables
    a: (0)
    b: (1)

  Instructions
    1: ADD (0) (0)

BB2 (outer header):
  Program Variables
    a: (7)
    b: (13)

  Instructions
    7: phi (6)  (0)
   13: phi (12) (1)
    3: CMP (7)  (2)
    4: BGE (3) (5)

BB3 (fall-through, BB2)
  Program Variables
    a: (6)

  Instructions
    6: ADD (7)  (0)
      <<trigger phi creation of (7) on BB2>>

BB4 (inner header):
  Program Variables
    b: (12)

  Instructions
    12: phi (11) (13)
     8: CMP (6)  (12)
     9: BGE (8)  (10)
  
BB5 (inner loop body)
  Program Variables:
    b (11)

  Instructions:
    11: Add (12) (0)
    <<trigger phi creation of (12) on BB4>>

BB6 (inner follow):
  10: NOP
    : BRA (7) 

<<trigger phi creation of (13) for b on BB2>>

BB7 (outside follow):
  Instructions
    5: NOP
